eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/configuration/ApplicationConfiguration.java

```diff
@@ -70,4 +86,24 @@ public void addCorsMappings(CorsRegistry registry) {
             }
         };
     }
+
+    /**
+     * Implementation reused from {@link WebClientAutoConfiguration}, but with an extra call to the method
+     * {@link org.springframework.boot.web.client.RestTemplateBuilder#requestFactory} to provide our
+     * {@link NonRedirectingClientHttpRequestFactory}.
+     */
+    @Bean
+    @ConditionalOnClass(RestTemplate.class)
```

Comment:
We have explicitly imported RestTemplate in the imports section. So, do we need the ConditionalOnClass here?

Reply:
ConditionalOnClass means that it will create this bean only if RestTemplate was requested as a bean (e.g. for autowiring). Unless I misunderstood the question, the imports section doesn't have anything to do with this.

Reply:
Ok, ConditionalOnClass, I confused it with ConditionalOnBean. still the original implementation does the same. worst case scenario: it's redundant.